src/tools/pmempool/common.c

```diff
@@ -915,11 +927,14 @@ util_opt_get_req(const struct options *opts, int opt, pmem_pool_type_t type)
 {
 	size_t n = 0;
 	struct option_requirement *ret = NULL;
+	struct option_requirement *tmp = NULL;
 	const struct option_requirement *req = &opts->req[0];
 	while (req->opt) {
 		if (req->opt == opt && (req->type & type)) {
 			n++;
-			ret = realloc(ret, n * sizeof(*ret));
+			tmp = realloc(ret, n * sizeof(*ret));
+			if (tmp)
```

Comment:
This is broken now. The second if will never be true on all loop invocations after the first one.
```
void *r = realloc(ptr, ...);
if (r == NULL) return;
ptr = r;

```